infra/presubmit.py

```diff
@@ -0,0 +1,173 @@
+#!/usr/bin/env python3
+# Copyright 2020 Google Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+################################################################################
+"""Statically check project for common issues."""
+
+from __future__ import print_function
+
+import itertools
+import os
+from multiprocessing.pool import ThreadPool
+import sys
+import yaml
+
+
+def get_oss_fuzz_root():
+  """Get the absolute path of the root of the oss-fuzz checkout."""
+  script_path = os.path.realpath(__file__)
+  return os.path.abspath(os.path.join(os.path.dirname(script_path), '..'))
+
+
+def is_project_file(actual_path, expected_filename):
+  if os.path.basename(actual_path) != expected_filename:
+    return False
+
+  if os.path.basename(
+      os.path.dirname(os.path.dirname(actual_path))) != 'projects':
+    return False
+
+  return os.path.exists(actual_path)
+
+
+def check_lib_fuzzing_engine(build_sh_file):
+  if not is_project_file(build_sh_file, 'build.sh'):
+    return True
+
+  with open(build_sh_file) as build_sh:
+    build_sh_lines = build_sh.readlines()
+  for line_num, line in enumerate(build_sh_lines):
+    uncommented_code = line.split('#')[0]
+    if '-lFuzzingEngine' in uncommented_code:
+      print('''Error: build.sh contains -lFuzzingEngine on line: {0}.
+Please use $LIB_FUZZING_ENGINE.'''.format(line_num))
+      return False
+  return True
+
+
+class ProjectYamlChecker:
+  def __init__(self, project_yaml_filename):
+    self.project_yaml_filename = project_yaml_filename
+    with open(project_yaml_filename) as file_handle:
+      self.project_yaml = yaml.safe_load(file_handle)
+
+    self.success = True
+
+    self.checks = [self.check_project_yaml_constants,
```

Comment:
no need for this to be an instance var? just initialise it in the do_checks function.